src/core/layout/qgslayoutitemlegend.cpp

```diff
@@ -845,48 +851,156 @@ QgsExpressionContext QgsLayoutItemLegend::createExpressionContext() const
   scope->addVariable( QgsExpressionContextScope::StaticVariable( QStringLiteral( "legend_filter_out_atlas" ), legendFilterOutAtlas(), true ) );
 
   context.appendScope( scope );
-
   return context;
 }
 
+QgsExpressionContext QgsLayoutItemLegend::updateExpressionContext()
+{
+
+  QgsExpressionContext context = createExpressionContext();
+
+  mExpContext.~QgsExpressionContext();
+  const QList<QgsExpressionContextScope *> scopes = context.takeScopes();
+  mExpContext.appendScopes( scopes );
+  return mExpContext;
+
+}
+
+
 // -------------------------------------------------------------------------
 #include "qgslayertreemodellegendnode.h"
 #include "qgsvectorlayer.h"
+#include "qgsmaplayerlegend.h"
 
 QgsLegendModel::QgsLegendModel( QgsLayerTree *rootNode, QObject *parent )
   : QgsLayerTreeModel( rootNode, parent )
 {
+  mLayoutLegendContext = nullptr;
   setFlag( QgsLayerTreeModel::AllowLegendChangeState, false );
   setFlag( QgsLayerTreeModel::AllowNodeReorder, true );
 }
 
+Qt::ItemFlags QgsLegendModel::flags( const QModelIndex &index ) const
```

Comment:
I think this is a leftover, no?

Reply:
This was originally here and I did not want to break the api by removing it, if you check the original was on line 885, below.

Reply:
Ah yes, I see :+1: 
Best to move it back there after the `data()` method so it doesn't clutter the diff here.

Reply:
Done